lib/sles4sap/ipaddr2.pm

```diff
@@ -765,26 +766,43 @@ die in case of failure
 sub ipaddr2_os_connectivity_sanity {
     my (%args) = @_;
     $args{bastion_ip} //= ipaddr2_bastion_pubip();
+    my $ping_cmd = 'ping -c 3';
 
     # proceed_on_failure needed as ping or nc
     # could be missing on the qcow2 running these commands
     # (for example pc_tools)
-    script_run("ping -c 3 $args{bastion_ip}", proceed_on_failure => 1);
+    script_run("$ping_cmd $args{bastion_ip}", proceed_on_failure => 1);
     script_run("nc -vz -w 1 $args{bastion_ip} 22", proceed_on_failure => 1);
 
-    # Check if the bastion is able to ping
-    # the VM by hostname and private IP
+
     foreach my $i (1 .. 2) {
+        # Check if the bastion is able to ping
+        # the VM by hostname and private IP
         foreach my $addr (
             ipaddr2_get_internal_vm_private_ip(id => $i),
             ipaddr2_get_internal_vm_name(id => $i)) {
-            foreach my $cmd ('ping -c 3 ', 'tracepath ', 'dig ') {
+            foreach my $cmd ($ping_cmd, 'tracepath', 'dig') {
                 ipaddr2_ssh_bastion_assert_script_run(
                     cmd => "$cmd $addr",
                     bastion_ip => $args{bastion_ip});
             }
         }
+        # Check if each internal VM can ping the virtual IP
+        ipaddr2_ssh_internal(
+            id => $i,
+            cmd => join(' ', $ping_cmd, $frontend_ip),
+            bastion_ip => $args{bastion_ip});
     }
+
+    ipaddr2_ssh_internal(
+        id => 1,
+        cmd => join(' ', $ping_cmd, ipaddr2_get_internal_vm_private_ip(id => 2)),
+        bastion_ip => $args{bastion_ip});
+
+    ipaddr2_ssh_internal(
+        id => 2,
+        cmd => 'ping -c 3 ' . ipaddr2_get_internal_vm_private_ip(id => 1),
```

Comment:
no problem for this line, but it is better if the format of this line is the same with line#799.
and is it possible to move the current `ipaddr2_ssh_internal` (two functions) into the above if-statement `foreach my $i (1 .. 2) {` with the variable `id` ?

Reply:
> it is better if the format of this line is the same with line#799.

ok, thanks. I'm going to update 

> possible to move the current ipaddr2_ssh_internal (two functions) into the above if-statement 

maybe but I do not know how to do it elegantly, those two commands are about testing if node 1 can ping node 2 and viceversa. So one call has to be  like `ping(1,2)` and the other `ping(2,1)`. The foreach is over 1...2 but I do not want node1 to try to ping itself...

Reply:
> maybe but I do not know how to do it elegantly, those two commands are about testing if node 1 can ping node 2 and viceversa. So one call has to be like ping(1,2) and the other ping(2,1). The foreach is over 1...2 but I do not want node1 to try to ping itself...
I see, please handle it the way you like :smile: